### PR TITLE
Add threading to host source reads

### PR DESCRIPTION
**Pull request by JKHiggins**

### CODE REVIEW

File: lib/interferon.rb [hunk 1]
```diff
@@ -149,15 +150,14 @@ def read_groups(sources)
     def read_hosts(sources)
       statsd.gauge('hosts.sources', sources.count)
 
-      hosts = []
       loader = HostSourcesLoader.new([@alerts_repo_path])
-      loader.get_all(sources).each do |source|
+      hosts = Parallel.map(loader.get_all(sources), in_threads: @read_hosts_threads) do |source|
         break if @request_shutdown
```
Comment (jimmyngo) [inline lib/interferon.rb]: I believe this needs to be changed to `raise Parallel::Kill if @request_shutdown`

**Comment by JKHiggins** [comment]
Yeah, we tested with running per host and with it being configurable and found that per host still runs as slow as the slowest, so re-using threads is probably still ideal.

Also, consumers of interferon could have arbitrarily many hosts and may be running on a box that could support more or less threads than the number of hosts. Because of that, leaving it as configurable seems to make the most sense to me.

File: lib/interferon.rb (hunk 1 above)
Comment (JKHiggins) [reply lib/interferon.rb]: Yeah, I think you're right. I'll make this change.

**Comment by jimmyngo** [comment]
@JKHiggins I just finished merging all the changes I had, can we start rebasing some of these?

**Comment by JKHiggins** [comment]
@jimmyngo merge conflicts merged on this one! Thanks a bunch for getting back to us on this. :)

**Pull request merged**